### DISCUSSION TITLE
Console.error when API fails

### DISCUSSION THREAD

**Discussion by ninobc** [TanStack/query#622] [General] [answered]
**Describe the bug**
In both `useQuery` and `useMutation`, whenever the API that has been passed to them fails, it results in a console.error which would then appear on screen. Also, when the mutation API fails, it crashes the app entirely and the error can't be caught, even if calling the mutation within a try/catch block as mentioned in the docs.

**Additional context**
using it in a react native app.

**Comment by tannerlinsley** [comment, marked as answer]
When using in React Native, you'll need to override the Logger, otherwise, it will cause redboxing all over the place.

```js
import { setConsole } from 'react-query'

setConsole({
  log: console.log,
  warn: console.log,
  error: console.log,
})
```

**Reply by ayush987goyal** [reply]
This was really helpful. We probably might need to add it to the docs somewhere. Hi @tannerlinsley! Where, in the docs, do you think this should go? I can create a PR for the same.

**Reply by kopax** [reply]
This doesn't exist in react query anymore, what's the solution now ?

**Reply by TkDodo** [reply]
`setConsole` has been replaced with `setLogger`. Have you seen the v3 migration guide? https://react-query.tanstack.com/guides/migrating-to-react-query-3#setconsole-has-been-replaced-by-the-new-setlogger-function

**Reply by kopax** [reply]
Yes, sorry I posted that a minute before, all good and thanks !

**Reply by rshettyawaken** [reply]
Since, passing the custom logger is deprecated, how can I avoid yellow box for error responses in React Native?

**Reply by TkDodo** [reply]
we won't log error responses to the console anymore, so you should only see a log statement from react-query when there is really something that needs action from you as a developer.

**Comment by ninobc** [comment]
Just what I needed. Thanks a ton 🙌

**Comment by neilakoh** [comment]
I've been scratching my head for hours thinking what went wrong... adding it in the documentation would be a lot better or just add it by default
